### PR TITLE
Fix avatar group dependencies without effect behivour

### PR DESCRIPTION
**Pull request by hadasfa**
https://user-images.githubusercontent.com/72390374/186352970-c21718ef-6700-42ac-8e35-21ff33408c76.mov

### CODE REVIEW

File: src/components/AvatarGroup/AvatarGroupCounter.jsx [hunk 1]
```diff
@@ -36,7 +36,7 @@ const AvatarGroupCounter = ({
   const focusPrevPlaceholderRef = useRef(null);
   const focusNextPlaceholderRef = useRef(null);
   const counterContainerRef = useRef(null);
-
+  console.log("mewo", counterContainerRef);
```
Comment (orrgottlieb) [inline src/components/AvatarGroup/AvatarGroupCounter.jsx]: remove please 🙏

File: src/components/AvatarGroup/AvatarGroupCounterTooltipContainer.jsx [hunk 2]
```diff
@@ -19,10 +19,22 @@ const AvatarGroupCounterTooltipContainer = ({
...
+    hide: [Dialog.hideShowTriggers.MOUSE_LEAVE]
+  };
+
   // Used to close tooltip
-  const [isTooltipVisible, setIsTooltipVisible] = useState(true);
+  const [isKeyboardTooltipVisible, setIsKeyboardTooltipVisible] = useState(false);
 
   const tooltipContentContainerRef = useRef(null);
+  useEffect(() => {
+    console.log(counterContainerRef);
```
Comment (orrgottlieb) [inline src/components/AvatarGroup/AvatarGroupCounterTooltipContainer.jsx]: remove console.log

File: src/components/AvatarGroup/AvatarGroupCounterTooltipContainer.jsx [hunk 3]
```diff
@@ -19,10 +19,22 @@ const AvatarGroupCounterTooltipContainer = ({
 }) => {
   // Dummy state to rerender the component, when tooltip appear, so useTooltipContentTabNavigation will have an existing tooltipContentContainerRef
   const [, setShouldUpdate] = useState(false);
+  const keyboardTooltipTrigger = {
```
Comment (orrgottlieb) [inline src/components/AvatarGroup/AvatarGroupCounterTooltipContainer.jsx:22]: please either constant outside of the function or useMemo

File: src/components/AvatarGroup/AvatarGroupCounterTooltipContainer.jsx [hunk 4]
```diff
@@ -43,23 +55,27 @@ const AvatarGroupCounterTooltipContainer = ({
...
+  }, []);
+
   if (!avatars?.length && !counterTooltipCustomProps?.content) {
     return children;
   }
 
-  if (!isTooltipVisible) {
-    return children;
-  }
+  const showTrigger = [...keyboardTooltipTrigger.show, ...mouseTooltipTrigger.show];
```
Comment (orrgottlieb) [inline src/components/AvatarGroup/AvatarGroupCounterTooltipContainer.jsx:67]: memoize both please

**Review by orrgottlieb — CHANGES_REQUESTED** [review]

File: src/components/AvatarGroup/AvatarGroupCounterTooltipHelper.js [hunk 5]
```diff
@@ -4,88 +4,103 @@ import Flex from "../Flex/Flex";
...
-      e => {
-        if (e.target === tooltipContentContainerRef.current) {
-          focusNextPlaceholderRef?.current && focusNextPlaceholderRef.current.focus();
-          hideTooltip();
-        }
-      },
-      [focusNextPlaceholderRef, hideTooltip, tooltipContentContainerRef.current]
-    )
+    callback: useCallback(() => {
+      // We are not preventing default behaviour here and that's why after moving focus to here we
```
Comment (hadasfa) [inline src/components/AvatarGroup/AvatarGroupCounterTooltipHelper.js:68]:
```suggestion
      // We are not preventing default behaviour here and that's why after pressing tab and after moving focus to here
```

File: src/components/AvatarGroup/AvatarGroupCounterTooltipHelper.js [hunk 6]
```diff
@@ -4,88 +4,103 @@ import Flex from "../Flex/Flex";
...
-      e => {
-        if (e.target === tooltipContentContainerRef.current) {
-          e.preventDefault();
-          counterContainerRef?.current && counterContainerRef.current.focus();
-        }
-      },
-      [counterContainerRef, tooltipContentContainerRef.current]
-    )
+    callback: useCallback(() => {
+      // We are not preventing default behaviour here and that's why after moving focus to here we
```
Comment (hadasfa) [inline src/components/AvatarGroup/AvatarGroupCounterTooltipHelper.js:79]:
```suggestion
      // We are not preventing default behaviour here and that's why after pressing tab and after moving focus to here
```

File: src/components/AvatarGroup/AvatarGroupCounterTooltipContainer.jsx [hunk 7]
```diff
@@ -42,24 +38,29 @@ const AvatarGroupCounterTooltipContainer = ({
...
 
-  if (!avatars?.length && !counterTooltipCustomProps?.content) {
-    return children;
-  }
+  // Tooltip props
+  const onHide = useCallback(() => {
+    setIsKeyboardTooltipVisible(false);
+  }, []);
+  const showTrigger = useMemo(() => [Dialog.hideShowTriggers.MOUSE_ENTER], []);
+  const hideTrigger = useMemo(() => [Dialog.hideShowTriggers.MOUSE_LEAVE], []);
```
Comment (SergeyRoyt) [inline src/components/AvatarGroup/AvatarGroupCounterTooltipContainer.jsx:50]: Why we can't just define these consts outside of component declaration without using memo?

File: src/components/AvatarGroup/AvatarGroupCounterTooltipContainer.jsx [hunk 8]
```diff
@@ -42,24 +38,29 @@ const AvatarGroupCounterTooltipContainer = ({
...
-  if (!isTooltipVisible) {
+  if (!avatars?.length && !counterTooltipCustomProps?.content) {
     return children;
   }
-
   return (
     <Tooltip
+      // for disable close tooltip when hovering content
       showOnDialogEnter
+      open={isKeyboardTooltipVisible}
```
Comment (SergeyRoyt) [inline src/components/AvatarGroup/AvatarGroupCounterTooltipContainer.jsx:59]: awesome! I've missed this prop somehow

File: src/components/AvatarGroup/AvatarGroupCounterTooltipHelper.js [hunk 9]
```diff
@@ -4,88 +4,103 @@ import Flex from "../Flex/Flex";
...
 }) {
-  const hideTooltip = useCallback(() => {
-    // Tricky way to close the tooltip
-    setTimeout(() => {
-      setIsTooltipVisible(false);
-      setIsTooltipVisible(true);
-    });
-  }, [setIsTooltipVisible]);
+  const showKeyboardTooltip = useCallback(() => {
+    if (!isKeyboardTooltipVisible) setIsKeyboardTooltipVisible(true);
```
Comment (SergeyRoyt) [inline src/components/AvatarGroup/AvatarGroupCounterTooltipHelper.js:22]: There is an eye-catchable difference between tooltip opening for Avatar and AvatarGroupCounter - for AvatarGroupCounter it's opening without delay. Let's try to make it consistent

File: src/components/AvatarGroup/AvatarGroupCounterTooltipHelper.js [hunk 10]
```diff
@@ -4,88 +4,103 @@ import Flex from "../Flex/Flex";
...
     ref: counterContainerRef,
+    withoutAnyModifier: true,
     callback: useCallback(
       e => {
-        if (e.target === counterContainerRef.current) {
-          e.preventDefault();
-          tooltipContentContainerRef?.current && tooltipContentContainerRef.current.focus();
-          setShouldUpdate(prev => !prev);
-        }
+        e.preventDefault();
```
Comment (SergeyRoyt) [inline src/components/AvatarGroup/AvatarGroupCounterTooltipHelper.js:49]: No condition here, so it's better to put `preventDefault` in the useKeyEvent props

File: src/components/AvatarGroup/AvatarGroupCounterTooltipContainer.jsx (hunk 7 above)
Comment (hadasfa) [reply src/components/AvatarGroup/AvatarGroupCounterTooltipContainer.jsx:50]: Or request here a memo... I thought that in this case there is not matter for giving to dialog component a new instance each render. wdyt @orrgottlieb ?

**Review by SergeyRoyt — APPROVED** [review]
👍

**Pull request merged**